Golem/Jobs.cs

```diff
@@ -95,16 +97,82 @@ public void UpdatePaymentConfirmation(string jobId, List<Payment> payments)
             _logger.LogInformation("Payments confirmation for job {0}:", job.Id);
 
             job.PaymentConfirmation = payments;
+            job.OnPropertyChanged();
         }
         else
         {
             _logger.LogWarning($"Failed to update payment confirmation. Job not found: {jobId}");
         }
     }
 
-    public Task<List<IJob>> List()
+    public async Task<List<IJob>> List(DateTime since)
     {
-        return Task.FromResult(_jobs.Values.Select(job => job as IJob).ToList());
+        if(_yagna == null || _yagna.HasExited)
+            return new List<IJob>();
+        var activities = await _yagna.Api.GetActivities();
+        var invoices = await _yagna.Api.GetInvoices(since);
+        var payments = await _yagna.Api.GetPayments(since);
+
+        var activityAgreements = new Dictionary<string, YagnaAgreement>();
+        var activityStates = new Dictionary<string, ActivityState>();
+
+        foreach(var activityId in activities)
+        {
+            var agreementId = await _yagna.Api.GetActivityAgreement(activityId);
+            
+            var (agreement, state) = await GetAgreementAndState(agreementId, activityId);
+
+            if(agreement == null || state == null)
+                continue;
+
+            if (agreement.Timestamp < since)
+                continue;
+
+            activityAgreements[activityId] = agreement;
+
+            activityStates[activityId] = new ActivityState
+            {
+                AgreementId = agreementId,
+                Id = activityId,
+                State = state.currentState()
+            };
+        }
+
+        foreach(var activityAgreement in activityAgreements)
+        {    
+            var agreement = activityAgreement.Value;
+            var activityId = activityAgreement.Key;
+        
+            if(agreement.AgreementID == null || agreement.Demand?.RequestorId == null)
+                continue;
+
+            var agreementId = agreement.AgreementID;
+
+            if(!_jobs.ContainsKey(agreementId))
+            {
+                _jobs[agreementId] = GetOrCreateJob(agreementId, agreement);
+            }
+            var job = _jobs[agreementId];
```

Comment:
This check doesn't make sense, because already does this and returns Job which you want